lib/routers/establishment/index.js

```diff
@@ -36,8 +37,12 @@ router.param('establishment', (req, res, next, id) => {
   Promise.resolve()
     .then(() => {
       return Establishment.query()
+        .select('establishments.*')
+        .count('places.id', {as: 'placesCount'})
+        .leftJoin('places', 'places.establishment_id', 'establishments.id')
         .findById(id)
-        .eager('[authorisations, roles.profile, asru]');
+        .eager('[authorisations, roles.profile, asru]')
+        .groupBy('establishments.id');
```

Comment:
I'd be _tempted_ to do this in a separate query, purely to assist SQL n00bs like me understand what's going on.

It doesn't really matter though.

Reply:
We could add a second query to add the `placesCount` prop, but it'd still be best to use `count()` & `groupBy()` because returning all the places data just to count them, e.g. 
```
Places.query().where({establishmentId: establishment.id})
  .then(places => places.length);
```
is crazy inefficient if there are lots of places.

So with that in mind it doesn't feel much simpler to separate them?

Reply:
Yeah, it was mostly the `groupBy` that I didn't really get. It seemed redundant on account of there only being one establishment. It's not clear what it's _for_.

Definitely wouldn't do the "get them all and read the length" thing though, that makes absolute sense.

Leave it as it is. It'll just be on me to understand databases better. I did once upon a time in a past life, so I'm sure I can again.